rpc/namespaces/ethereum/eth/api.go

```diff
@@ -825,8 +841,17 @@ func (e *PublicAPI) GetTransactionReceipt(hash common.Hash) (map[string]interfac
 		return nil, nil
 	}
 
-	msgIndex, attrs := rpctypes.FindTxAttributes(res.TxResult.Events, hexTx)
-	if msgIndex < 0 {
+	if !backend.ShouldIncludeInJSONRPC(&res.TxResult) {
```

Comment:
this function is not very descriptive. Shouldn't we mention when the tx results are excluded?

Reply:
maybe sth like `TxSuccessOrExceedsBlockGasLimit`?